environment/src/main/java/com/sequenceiq/environment/environment/validation/validators/EncryptionKeyArnValidator.java

```diff
@@ -25,4 +59,32 @@ public ValidationResult validateEncryptionKeyArn(String encryptionKeyArn) {
         }
         return validationResultBuilder.build();
     }
+
+    public ValidationResult validate(EnvironmentValidationDto environmentValidationDto, ParametersDto parametersDto) {
+        EnvironmentDto environmentDto = environmentValidationDto.getEnvironmentDto();
+        ValidationResult.ValidationResultBuilder validationResultBuilder = ValidationResult.builder();
+        String encryptionKeyArn = parametersDto.getAwsParametersDto().getAwsDiskEncryptionParametersDto().getEncryptionKeyArn();
+        CloudCredential cloudCredential = credentialToCloudCredentialConverter.convert(environmentDto.getCredential());
+        Region region = region(environmentDto.getLocation().getName());
+        new CloudEncryptionKeys(new HashSet<>());
+        CloudEncryptionKeys encryptionKeys;
+        CloudPlatformVariant cloudPlatformVariant = new CloudPlatformVariant(
+                Platform.platform(environmentDto.getCloudPlatform()), null);
+        try {
+            encryptionKeys =  retryService.testWith2SecDelayMax15Times(() -> cloudPlatformConnectors.get(cloudPlatformVariant).
+                                                                            platformResources().encryptionKeys(cloudCredential, region, Collections.emptyMap()));
+        } catch (Retry.ActionFailedException | AmazonServiceException e) {
+            LOGGER.error("An unexpected error occurred while trying to fetch the KMS keys from AWS");
+            throw e;
+        }
```

Comment:
Let's go with this for now, but needs unit testing.

Reply:
Added the unit tests